pages/_app.tsx

```diff
@@ -1,37 +1,42 @@
 // Patternfly import needs to be first or the css will not be imported
 import '@patternfly/react-core/dist/styles/base.css';
+import type { SegmentEvent } from '@segment/analytics-next';
 import { Layout } from '@src/components';
-import { getAnalytics, getUserRegion } from '@src/util/client';
+import { getUserRegion, useAnalytics } from '@src/util/client';
 import { AppProps } from 'next/app';
 import { useRouter } from 'next/router';
 import { NextPage } from 'next';
-import { useEffect } from 'react';
 import getConfig from 'next/config';
+import { useEffect } from 'react';
 
 /**
  * Renders the {@link MyApp}
  */
 const MyApp: NextPage<AppProps> = ({ Component, pageProps }: AppProps) => {
   const router = useRouter();
+  const analytics = useAnalytics();
 
   // Client renders page
   useEffect(() => {
-    const analytics = getAnalytics();
     const region = getUserRegion(router.locale);
     const { publicRuntimeConfig } = getConfig();
 
     if (analytics) {
-      analytics.page({
-        userId: publicRuntimeConfig.segmentUserId,
+      const event: SegmentEvent = {
+        type: 'track',
+        anonymousId: analytics.user().anonymousId(),
```

Comment:
Is there any situation where this can return `nil` or empty?  Just wondering what happens if the user has an ad blocker or disables cookie tracking, can we still generate an anonymous ID?   If not, we should default to the generic `client` name if that's the case